build: Drop unneeded dependencies for bitcoin-wallet tool

hebasto opened the pull request:
`bitcoin-wallet` is an offline tool, and its code does not depend on networking stuff (ZMQ, UPnP, NAT-PMP, and LIBBITCOIN_SERVER).

Also `bitcoin-wallet`  does not interacts with the chainstate, therefore dependency on LevelDB is not needed.

hebasto committed:
build: Drop unneeded dependencies for bitcoin-wallet tool

luke-jr reviewed (commented):
Doesn't this also need configure changes to allow building it without unnecessary deps? (Possibly could be a followup PR, but fits here)

hebasto commented:
Updated 1d96788ce97f3dcfbe326652c0d6204624dbb783 -> 347774b86c8e451300c9487d8c350ebdb0df2f86 ([pr23345.01](https://github.com/hebasto/bitcoin/commits/pr23345.01) -> [pr23345.02](https://github.com/hebasto/bitcoin/commits/pr23345.02), [diff](https://github.com/hebasto/bitcoin/compare/pr23345.01..pr23345.02)):

- addressed @luke-jr's [comment](https://github.com/bitcoin/bitcoin/pull/23345#pullrequestreview-793266469):

> Doesn't this also need configure changes to allow building it without unnecessary deps?

hebasto commented:
#### Guix builds:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
f3e5d495458d3347503f3446b6810b75bc519c5508986f5225e1ee8e3187aead  guix-build-347774b86c8e/output/aarch64-linux-gnu/SHA256SUMS.part
6d8dcea67a1295cf04f121697779fb1158e74fed6ec317cd81f0e5a10f82d532  guix-build-347774b86c8e/output/aarch64-linux-gnu/bitcoin-347774b86c8e-aarch64-linux-gnu-debug.tar.gz
c18359b4048a113d984d3a1bf67d19c25d9b27513f5b112e8359289f6d3ca907  guix-build-347774b86c8e/output/aarch64-linux-gnu/bitcoin-347774b86c8e-aarch64-linux-gnu.tar.gz
8cb85a9815cd5046aa38db22bd8593d5e09ffa09935a7008dda722174aa4343b  guix-build-347774b86c8e/output/arm-linux-gnueabihf/SHA256SUMS.part
cf549a6d76361be67f40acb227e29c1cd1503bf71c750a80c18cb8f99945085d  guix-build-347774b86c8e/output/arm-linux-gnueabihf/bitcoin-347774b86c8e-arm-linux-gnueabihf-debug.tar.gz
9b15e710313cefa6e2e9ee7ac7d7945b55399c6ac2ba73a24565393fb1eceded  guix-build-347774b86c8e/output/arm-linux-gnueabihf/bitcoin-347774b86c8e-arm-linux-gnueabihf.tar.gz
9bd9aecf8c6f47e04812a2604bdbd59b51a4982f2a9665ecae3d8ed4ad174ac2  guix-build-347774b86c8e/output/dist-archive/bitcoin-347774b86c8e.tar.gz
d3021be3d52aac1578839361baa657175f2f95a6a29db149b14782ad7321051b  guix-build-347774b86c8e/output/powerpc64-linux-gnu/SHA256SUMS.part
8087e54dc2123f68ef0be25035a759b32e30bc986c371566636227c500748af2  guix-build-347774b86c8e/output/powerpc64-linux-gnu/bitcoin-347774b86c8e-powerpc64-linux-gnu-debug.tar.gz
4d48a90cd9946d76f3f27ce258f2d693b6e5b9aa35d5027b9f599d8ad6e95245  guix-build-347774b86c8e/output/powerpc64-linux-gnu/bitcoin-347774b86c8e-powerpc64-linux-gnu.tar.gz
d8bbf33587375528e26ec199242d94a198e59803f496abcf61de0a2ebccbdf02  guix-build-347774b86c8e/output/powerpc64le-linux-gnu/SHA256SUMS.part
2a2eecd98b92f90137cc9f6c309999b369c1bf7de725d91c345b5efe6c46b801  guix-build-347774b86c8e/output/powerpc64le-linux-gnu/bitcoin-347774b86c8e-powerpc64le-linux-gnu-debug.tar.gz
d7405a053343d63882d3fd67ae216bf90c3f41c09220e4703bf0badd30850863  guix-build-347774b86c8e/output/powerpc64le-linux-gnu/bitcoin-347774b86c8e-powerpc64le-linux-gnu.tar.gz
8514715838bf9564479dc159a7f0c33fa764deb1ff017593554f37af5da3df09  guix-build-347774b86c8e/output/riscv64-linux-gnu/SHA256SUMS.part
a675d706bec9a78fce3062db293501ac787e05aa606d41c3836dc5bf3a47a8ef  guix-build-347774b86c8e/output/riscv64-linux-gnu/bitcoin-347774b86c8e-riscv64-linux-gnu-debug.tar.gz
d27910d10af4d0515d5d1568bf289ef0d9aff954d1410d0c695322a36fb3f9a0  guix-build-347774b86c8e/output/riscv64-linux-gnu/bitcoin-347774b86c8e-riscv64-linux-gnu.tar.gz
a8445bb3a96cb57de9369722092a887843afd58747dca1a2442529bced039a32  guix-build-347774b86c8e/output/x86_64-apple-darwin19/SHA256SUMS.part
c3cc41b10cbd6caf3faa9ef2a6d537f2401df5daaea40a907315fd7ee4243d6d  guix-build-347774b86c8e/output/x86_64-apple-darwin19/bitcoin-347774b86c8e-osx-unsigned.dmg
97c94bf214a40f86a71fb5860142ab08e2a8e21dd4278abe8abd4aa6a92664c2  guix-build-347774b86c8e/output/x86_64-apple-darwin19/bitcoin-347774b86c8e-osx-unsigned.tar.gz
f9d73b98b424b0e4ece233759d19370452dbbe50e635a13ede2c2a93e737c284  guix-build-347774b86c8e/output/x86_64-apple-darwin19/bitcoin-347774b86c8e-osx64.tar.gz
eebf4699ef0a9971a2f44ccd996a696e163b8536a7f17ee9fa6de7f2dfc434f9  guix-build-347774b86c8e/output/x86_64-linux-gnu/SHA256SUMS.part
52736eddc2b240d8a666f251264bc83a1c3f73ce6c0de0a353263f3fb8e4baad  guix-build-347774b86c8e/output/x86_64-linux-gnu/bitcoin-347774b86c8e-x86_64-linux-gnu-debug.tar.gz
fec9b38aad06b47797f458fb97571d49dbce17f2023fee60517600f16ef352fd  guix-build-347774b86c8e/output/x86_64-linux-gnu/bitcoin-347774b86c8e-x86_64-linux-gnu.tar.gz
120a64e96ff9312212d29b7894e049f734e349d58b4424a40a0dbf1a81ef9f63  guix-build-347774b86c8e/output/x86_64-w64-mingw32/SHA256SUMS.part
66f22e1abc8278df1a2c31e97adb97567267b7f1df9fe3c54d497e8cadeda620  guix-build-347774b86c8e/output/x86_64-w64-mingw32/bitcoin-347774b86c8e-win-unsigned.tar.gz
28cd8f5f87bb27811d9d32542c93b3d343f84a39c20fa2cd0fc5f24ccd3bd924  guix-build-347774b86c8e/output/x86_64-w64-mingw32/bitcoin-347774b86c8e-win64-debug.zip
e9f4fed1b34dc6bec4130a40d62aa9bb1d02bee29a93a92c25a2278ae88d0c14  guix-build-347774b86c8e/output/x86_64-w64-mingw32/bitcoin-347774b86c8e-win64-setup-unsigned.exe
cef9f80f3a6541b66054068d244e64b49f13106f92099fe78d889e422e5bab21  guix-build-347774b86c8e/output/x86_64-w64-mingw32/bitcoin-347774b86c8e-win64.zip
```

jarolrod commented:
contributing GUIX hashes, mine match @hebasto 

```
find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum

f3e5d495458d3347503f3446b6810b75bc519c5508986f5225e1ee8e3187aead  guix-build-347774b86c8e/output/aarch64-linux-gnu/SHA256SUMS.part
6d8dcea67a1295cf04f121697779fb1158e74fed6ec317cd81f0e5a10f82d532  guix-build-347774b86c8e/output/aarch64-linux-gnu/bitcoin-347774b86c8e-aarch64-linux-gnu-debug.tar.gz
c18359b4048a113d984d3a1bf67d19c25d9b27513f5b112e8359289f6d3ca907  guix-build-347774b86c8e/output/aarch64-linux-gnu/bitcoin-347774b86c8e-aarch64-linux-gnu.tar.gz
8cb85a9815cd5046aa38db22bd8593d5e09ffa09935a7008dda722174aa4343b  guix-build-347774b86c8e/output/arm-linux-gnueabihf/SHA256SUMS.part
cf549a6d76361be67f40acb227e29c1cd1503bf71c750a80c18cb8f99945085d  guix-build-347774b86c8e/output/arm-linux-gnueabihf/bitcoin-347774b86c8e-arm-linux-gnueabihf-debug.tar.gz
9b15e710313cefa6e2e9ee7ac7d7945b55399c6ac2ba73a24565393fb1eceded  guix-build-347774b86c8e/output/arm-linux-gnueabihf/bitcoin-347774b86c8e-arm-linux-gnueabihf.tar.gz
9bd9aecf8c6f47e04812a2604bdbd59b51a4982f2a9665ecae3d8ed4ad174ac2  guix-build-347774b86c8e/output/dist-archive/bitcoin-347774b86c8e.tar.gz
d3021be3d52aac1578839361baa657175f2f95a6a29db149b14782ad7321051b  guix-build-347774b86c8e/output/powerpc64-linux-gnu/SHA256SUMS.part
8087e54dc2123f68ef0be25035a759b32e30bc986c371566636227c500748af2  guix-build-347774b86c8e/output/powerpc64-linux-gnu/bitcoin-347774b86c8e-powerpc64-linux-gnu-debug.tar.gz
4d48a90cd9946d76f3f27ce258f2d693b6e5b9aa35d5027b9f599d8ad6e95245  guix-build-347774b86c8e/output/powerpc64-linux-gnu/bitcoin-347774b86c8e-powerpc64-linux-gnu.tar.gz
d8bbf33587375528e26ec199242d94a198e59803f496abcf61de0a2ebccbdf02  guix-build-347774b86c8e/output/powerpc64le-linux-gnu/SHA256SUMS.part
2a2eecd98b92f90137cc9f6c309999b369c1bf7de725d91c345b5efe6c46b801  guix-build-347774b86c8e/output/powerpc64le-linux-gnu/bitcoin-347774b86c8e-powerpc64le-linux-gnu-debug.tar.gz
d7405a053343d63882d3fd67ae216bf90c3f41c09220e4703bf0badd30850863  guix-build-347774b86c8e/output/powerpc64le-linux-gnu/bitcoin-347774b86c8e-powerpc64le-linux-gnu.tar.gz
8514715838bf9564479dc159a7f0c33fa764deb1ff017593554f37af5da3df09  guix-build-347774b86c8e/output/riscv64-linux-gnu/SHA256SUMS.part
a675d706bec9a78fce3062db293501ac787e05aa606d41c3836dc5bf3a47a8ef  guix-build-347774b86c8e/output/riscv64-linux-gnu/bitcoin-347774b86c8e-riscv64-linux-gnu-debug.tar.gz
d27910d10af4d0515d5d1568bf289ef0d9aff954d1410d0c695322a36fb3f9a0  guix-build-347774b86c8e/output/riscv64-linux-gnu/bitcoin-347774b86c8e-riscv64-linux-gnu.tar.gz
a8445bb3a96cb57de9369722092a887843afd58747dca1a2442529bced039a32  guix-build-347774b86c8e/output/x86_64-apple-darwin19/SHA256SUMS.part
c3cc41b10cbd6caf3faa9ef2a6d537f2401df5daaea40a907315fd7ee4243d6d  guix-build-347774b86c8e/output/x86_64-apple-darwin19/bitcoin-347774b86c8e-osx-unsigned.dmg
97c94bf214a40f86a71fb5860142ab08e2a8e21dd4278abe8abd4aa6a92664c2  guix-build-347774b86c8e/output/x86_64-apple-darwin19/bitcoin-347774b86c8e-osx-unsigned.tar.gz
f9d73b98b424b0e4ece233759d19370452dbbe50e635a13ede2c2a93e737c284  guix-build-347774b86c8e/output/x86_64-apple-darwin19/bitcoin-347774b86c8e-osx64.tar.gz
eebf4699ef0a9971a2f44ccd996a696e163b8536a7f17ee9fa6de7f2dfc434f9  guix-build-347774b86c8e/output/x86_64-linux-gnu/SHA256SUMS.part
52736eddc2b240d8a666f251264bc83a1c3f73ce6c0de0a353263f3fb8e4baad  guix-build-347774b86c8e/output/x86_64-linux-gnu/bitcoin-347774b86c8e-x86_64-linux-gnu-debug.tar.gz
fec9b38aad06b47797f458fb97571d49dbce17f2023fee60517600f16ef352fd  guix-build-347774b86c8e/output/x86_64-linux-gnu/bitcoin-347774b86c8e-x86_64-linux-gnu.tar.gz
120a64e96ff9312212d29b7894e049f734e349d58b4424a40a0dbf1a81ef9f63  guix-build-347774b86c8e/output/x86_64-w64-mingw32/SHA256SUMS.part
66f22e1abc8278df1a2c31e97adb97567267b7f1df9fe3c54d497e8cadeda620  guix-build-347774b86c8e/output/x86_64-w64-mingw32/bitcoin-347774b86c8e-win-unsigned.tar.gz
28cd8f5f87bb27811d9d32542c93b3d343f84a39c20fa2cd0fc5f24ccd3bd924  guix-build-347774b86c8e/output/x86_64-w64-mingw32/bitcoin-347774b86c8e-win64-debug.zip
e9f4fed1b34dc6bec4130a40d62aa9bb1d02bee29a93a92c25a2278ae88d0c14  guix-build-347774b86c8e/output/x86_64-w64-mingw32/bitcoin-347774b86c8e-win64-setup-unsigned.exe
cef9f80f3a6541b66054068d244e64b49f13106f92099fe78d889e422e5bab21  guix-build-347774b86c8e/output/x86_64-w64-mingw32/bitcoin-347774b86c8e-win64.zip
```

luke-jr commented on configure.ac:
```diff
@@ -1336,6 +1336,15 @@ if test x$use_ebpf != xno; then
 fi
 
+build_leveldb=yes
+if test x$build_bitcoin_cli$build_bitcoin_tx$build_bitcoin_util$build_bitcoind$bitcoin_enable_qt$use_bench$use_tests = xnonononononono; then
```
I guess we should only be checking bitcoind/bitcoin-qt here (the other tools don't need them either?), but perhaps that's safer left for a later PR...

hebasto replied:
You read my mind :)

luke-jr replied:
Followup in #24051

laanwj commented on src/Makefile.am:
```diff
@@ -841,8 +852,11 @@ nodist_libbitcoin_ipc_a_SOURCES = $(libbitcoin_ipc_mpgen_output)
 endif
 
+if BUILD_LEVELDB
 include Makefile.crc32c.include
```
I don't understand why an explicit conditional is needed here. Shouldn't `make` be able to figure out for itself whether these libraries need to be built?

hebasto replied:
> Shouldn't `make` be able to figure out for itself whether these libraries need to be built?

Yes, it should.

> I don't understand why an explicit conditional is needed here.

It is not _needed_. It's an optimization -- not processing unused `Makefile.*.include` files.

hebasto replied:
[Updated](https://github.com/bitcoin/bitcoin/pull/23345#issuecomment-968125192).

laanwj replied:
Thanks. Yes, I don't think this was worth special-case optimizing, parsing performance of `make` is hardly an issue on modern systems afaik.

hebasto commented:
Updated 347774b86c8e451300c9487d8c350ebdb0df2f86 -> 4fe7cf167796015c0f24e740f81a2452e3996a92 ([pr23345.02](https://github.com/hebasto/bitcoin/commits/pr23345.02) -> [pr23345.03](https://github.com/hebasto/bitcoin/commits/pr23345.03)):
- rebased due to the conflict with #23114
- addressed @laanwj's [comment](https://github.com/bitcoin/bitcoin/pull/23345#discussion_r746860579)

hebasto commented:
#### Guix builds:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
f752d99d15d59e3a70a54cf73dd0d28c6f602a9c82f5303695618038e9370d77  guix-build-4fe7cf167796/output/aarch64-linux-gnu/SHA256SUMS.part
5857cc54913ab34737e269c2fd9ad29d32138f28436e11071bacf82f98d6c927  guix-build-4fe7cf167796/output/aarch64-linux-gnu/bitcoin-4fe7cf167796-aarch64-linux-gnu-debug.tar.gz
5d102280651e764d0f23cae6884813bf4f073ddbd966145a0fe56b2f560fae5d  guix-build-4fe7cf167796/output/aarch64-linux-gnu/bitcoin-4fe7cf167796-aarch64-linux-gnu.tar.gz
76f5453432a0c5d770034b40253ccbe17aa21e2c5dde30131c181988da21ad00  guix-build-4fe7cf167796/output/arm-linux-gnueabihf/SHA256SUMS.part
d8919579f061548f31588d3f519859ca1cfb8e92f7a627444218211be9f51972  guix-build-4fe7cf167796/output/arm-linux-gnueabihf/bitcoin-4fe7cf167796-arm-linux-gnueabihf-debug.tar.gz
3b30509b9deaff2a4ff63483afcc8b2418fc338ce962176cc4c4ed2cb6329c79  guix-build-4fe7cf167796/output/arm-linux-gnueabihf/bitcoin-4fe7cf167796-arm-linux-gnueabihf.tar.gz
244807167e14aff56d411028149afdc5ed052cb1ed4752ef6801158e3914961b  guix-build-4fe7cf167796/output/dist-archive/bitcoin-4fe7cf167796.tar.gz
4ced3674771be8c6f32d49e97511c6fd41c7f7f36412905f7e39d3e7e102663e  guix-build-4fe7cf167796/output/powerpc64-linux-gnu/SHA256SUMS.part
441b7fdcd38af2df3ecb44973b97b5011b09f7a01a69933bba01900073c9290e  guix-build-4fe7cf167796/output/powerpc64-linux-gnu/bitcoin-4fe7cf167796-powerpc64-linux-gnu-debug.tar.gz
697e0242b199370b6779f287dc38c988fb7e0ac09c89ab918daf7536fcb82d22  guix-build-4fe7cf167796/output/powerpc64-linux-gnu/bitcoin-4fe7cf167796-powerpc64-linux-gnu.tar.gz
c472ad36c754a8f7b446e687a7c214b411d55518991b25e058740fac3de2955a  guix-build-4fe7cf167796/output/powerpc64le-linux-gnu/SHA256SUMS.part
ee7eced6de534bd8fc50318436f0c73340ae0351f7e53073798b1999372f7c79  guix-build-4fe7cf167796/output/powerpc64le-linux-gnu/bitcoin-4fe7cf167796-powerpc64le-linux-gnu-debug.tar.gz
10abd0d49cb679ce49f009dff904f8210473f89b0a3299a9759f9545a40725d8  guix-build-4fe7cf167796/output/powerpc64le-linux-gnu/bitcoin-4fe7cf167796-powerpc64le-linux-gnu.tar.gz
21ba0c48fd660f7bb053fa09a97a888d764ee40eef26043e15682662f7e98e10  guix-build-4fe7cf167796/output/riscv64-linux-gnu/SHA256SUMS.part
46ccd9b042fffe532bbe7352c728aade6218444ade11a816d8a8e42ca3ad11eb  guix-build-4fe7cf167796/output/riscv64-linux-gnu/bitcoin-4fe7cf167796-riscv64-linux-gnu-debug.tar.gz
dd3a4e562f54a97c94c8b9be59aa123479d592cb2a08aa810df11a605a58b7bd  guix-build-4fe7cf167796/output/riscv64-linux-gnu/bitcoin-4fe7cf167796-riscv64-linux-gnu.tar.gz
6babf3b6a30d193f282dedefe94270ee286d9af5553fac3e6eb79391cf4e3cc0  guix-build-4fe7cf167796/output/x86_64-apple-darwin19/SHA256SUMS.part
eebcb0353d443ebfaea0ae1a78785205f1df57589e53c5d206a4ae8e1362ea76  guix-build-4fe7cf167796/output/x86_64-apple-darwin19/bitcoin-4fe7cf167796-osx-unsigned.dmg
2f8a09f1685bbead1cd26ac1f5037abee7efced27569f0f7c4fa39e852ebafca  guix-build-4fe7cf167796/output/x86_64-apple-darwin19/bitcoin-4fe7cf167796-osx-unsigned.tar.gz
29c4084fa88cd8bdcb13687f43f36c2ee5cadd672bbdb9a55c051158baa74179  guix-build-4fe7cf167796/output/x86_64-apple-darwin19/bitcoin-4fe7cf167796-osx64.tar.gz
2726aabb3e3ed9abd96da90f926aee9af5ccd4aeb1a27e94f67c55a94a121c08  guix-build-4fe7cf167796/output/x86_64-linux-gnu/SHA256SUMS.part
21712e64b021ee37621ec275e622999320a295533dc9b28c75702cce0a186ccd  guix-build-4fe7cf167796/output/x86_64-linux-gnu/bitcoin-4fe7cf167796-x86_64-linux-gnu-debug.tar.gz
f3f12e3fddf40660f2b6552eae1056897c837d4bcaa2e11835eb92055badc320  guix-build-4fe7cf167796/output/x86_64-linux-gnu/bitcoin-4fe7cf167796-x86_64-linux-gnu.tar.gz
84ad8907b64861c14c9635e93931d1f96c6690fb30ea029739d3107f6e7276c9  guix-build-4fe7cf167796/output/x86_64-w64-mingw32/SHA256SUMS.part
17c3e66ad6ac39a663121d4cdb6f9e37a098755a1dcaf770569ef018f7d5b796  guix-build-4fe7cf167796/output/x86_64-w64-mingw32/bitcoin-4fe7cf167796-win-unsigned.tar.gz
69f86bac1aba6742a3840256960feab3c14bd878517e714c7663c2c68c205c5e  guix-build-4fe7cf167796/output/x86_64-w64-mingw32/bitcoin-4fe7cf167796-win64-debug.zip
87af79229a21d407f68845767d420e8521d73e1103dae6359d846e6c070251ec  guix-build-4fe7cf167796/output/x86_64-w64-mingw32/bitcoin-4fe7cf167796-win64-setup-unsigned.exe
ee3a184ecbec4983e7bbdff29ab0ec46f7872f60f735df59984be141c516c215  guix-build-4fe7cf167796/output/x86_64-w64-mingw32/bitcoin-4fe7cf167796-win64.zip
```

laanwj commented:
Code review ACK 4fe7cf167796015c0f24e740f81a2452e3996a92
